crates/biome_css_parser/src/syntax/selector/mod.rs

```diff
@@ -5,37 +5,24 @@ mod pseudo_element;
 use crate::lexer::CssLexContext;
 use crate::parser::CssParser;
 use crate::syntax::parse_error::{
-    expect_any_selector, expect_any_sub_selector, expected_identifier,
+    expected_any_sub_selector, expected_compound_selector, expected_identifier, expected_selector,
 };
 use crate::syntax::selector::attribute::parse_attribute_selector;
 use crate::syntax::selector::pseudo_class::parse_pseudo_class_selector;
 use crate::syntax::selector::pseudo_element::parse_pseudo_element_selector;
 use crate::syntax::{is_at_identifier, parse_identifier, RULE_RECOVERY_SET};
 use biome_css_syntax::CssSyntaxKind::*;
-use biome_css_syntax::{CssSyntaxKind, T};
-use biome_parser::diagnostic::expected_token;
+use biome_css_syntax::{CssSyntaxKind, TextRange, T};
+use biome_parser::diagnostic::ToDiagnostic;
 use biome_parser::parse_lists::{ParseNodeList, ParseSeparatedList};
-use biome_parser::parse_recovery::{ParseRecovery, RecoveryResult};
+use biome_parser::parse_recovery::{ParseRecovery, RecoveryError, RecoveryResult};
 use biome_parser::prelude::ParsedSyntax;
 use biome_parser::prelude::ParsedSyntax::{Absent, Present};
 use biome_parser::{token_set, CompletedMarker, Parser, ParserProgress, TokenSet};
 
 const SELECTOR_RECOVERY_SET: TokenSet<CssSyntaxKind> = RULE_RECOVERY_SET.union(token_set![T![,]]);
 const SELECTOR_FUNCTION_RECOVERY_SET: TokenSet<CssSyntaxKind> = token_set![T![')'], T!['{']];
 
-#[inline]
-fn parse_selector_function_close_token(p: &mut CssParser) {
```

Comment:
I've changed a way how we recover if we don't find a close token.
I think that it's okey to consume all tokens until we find a `)`or  a `}`.
See snapshots for examples.